lib/srv/exec.go

```diff
@@ -481,27 +481,49 @@ func (r *remoteExec) collectRemoteStatus(err error) (*ExecResult, error) {
 }
 
 func emitExecAuditEvent(ctx *ServerContext, cmd string, status *ExecResult, err error) {
-	// report the result of this exec event to the audit logger
+	// Report the result of this exec event to the audit logger.
 	auditLog := ctx.srv.GetAuditLog()
 	if auditLog == nil {
 		log.Warnf("No audit log")
 		return
 	}
+
+	var eventType string
+
+	// Create common fields for event.
 	fields := events.EventFields{
-		events.ExecEventCommand: cmd,
-		events.EventUser:        ctx.Identity.TeleportUser,
-		events.EventLogin:       ctx.Identity.Login,
-		events.LocalAddr:        ctx.Conn.LocalAddr().String(),
-		events.RemoteAddr:       ctx.Conn.RemoteAddr().String(),
-		events.EventNamespace:   ctx.srv.GetNamespace(),
+		events.EventUser:      ctx.Identity.TeleportUser,
+		events.EventLogin:     ctx.Identity.Login,
+		events.LocalAddr:      ctx.Conn.LocalAddr().String(),
+		events.RemoteAddr:     ctx.Conn.RemoteAddr().String(),
+		events.EventNamespace: ctx.srv.GetNamespace(),
 	}
 	if err != nil {
 		fields[events.ExecEventError] = err.Error()
 		if status != nil {
 			fields[events.ExecEventCode] = strconv.Itoa(status.Code)
 		}
 	}
-	auditLog.EmitAuditEvent(events.ExecEvent, fields)
+
+	// Parse the exec command to find out if it was SCP or not.
+	path, action, isSCP, er := parseSecureCopy(cmd)
```

Comment:
let's refactor this a bit to stop using named returns, it introduces confusion, for example in this case function will return nil as an error, because `er` is used, but not `err`, however when I looked at the code calling it, the return value is ignored completely, so not sure why. I would like to simplify to alweays return the right error and let caller decide the fate

Reply:
This function does not used named returns, it doesn't return anything. I think the confusion is because the last argument to `emitExecAuditEvent` was `err error` which made it look like that.

I changed it to `execError error` which describes it better.